src/complex/Utilities/Parsing/DREAM3D/Dream3dIO.cpp

```diff
@@ -558,26 +594,25 @@ DataStructure ImportLegacyDataStructure(const H5::FileReader& fileReader, H5::Er
     readLegacyDataContainer(ds, dcGroup, preflight);
   }
 
-  return ds;
-
-  throw std::runtime_error("Not implemented: ImportLegacyDataStructure from dream3d file");
+  return {ds};
 }
 
-complex::DataStructure complex::DREAM3D::ImportDataStructureFromFile(const H5::FileReader& fileReader, H5::ErrorType& errorCode, bool preflight)
+Result<complex::DataStructure> complex::DREAM3D::ImportDataStructureFromFile(const H5::FileReader& fileReader, bool preflight)
 {
-  errorCode = 0;
-
   const auto fileVersion = GetFileVersion(fileReader);
   if(fileVersion == k_CurrentFileVersion)
   {
-    return ImportDataStructureV8(fileReader, errorCode, preflight);
+    return ImportDataStructureV8(fileReader, preflight);
   }
   else if(fileVersion == Legacy::FileVersion)
   {
-    return ImportLegacyDataStructure(fileReader, errorCode, preflight);
+    return ImportLegacyDataStructure(fileReader, preflight);
   }
   // Unsupported file version
-  return DataStructure();
+  Result<complex::DataStructure> result({});
+  result.warnings().push_back(
+      Warning{k_InvalidDataStructureVersion, fmt::format("Could not parse DataStructure version {}. Expected versions: {} or {}", fileVersion, k_CurrentFileVersion, Legacy::FileVersion)});
+  return result;
```

Comment:
```suggestion
  return MakeErrorResult<DataStructure>(k_InvalidDataStructureVersion, fmt::format("Could not parse DataStructure version {}. Expected versions: {} or {}", fileVersion, k_CurrentFileVersion, Legacy::FileVersion));
```
A default constructed result is valid.